ifrs17/Import/1ImportScope-Identities.ipynb

```diff
@@ -133,88 +141,25 @@
-                "\n                                                                                   DataNode = Identity }))",
-                "\n                                                                                   .Concat(new ImportIdentity {",
-                "\n                                                                                   AocType = AocTypes.CF, //Add CF for Deferral",
-                "\n                                                                                   Novelty = Novelties.C,",
-                "\n                                                                                   DataNode = Identity",
-                "\n                                                                                   }.RepeatOnce())   ",
-                "\n                                                                                    .Concat(GetStorage().GetAllAocSteps(InputSource.Opening)",
-                "\n                                                                                              .Select(aocStep => new ImportIdentity {AocType = aocStep.AocType, ",
-                "\n                                                                                                 Novelty = aocStep.Novelty,",
-                "\n                                                                                                 DataNode = Identity",
-                "\n                                                                                                 }));   ",
-                "\n}",
+                "\n    private bool isReinsurance => GetStorage().DataNodeDataBySystemName[Identity].IsReinsurance;",
+                "\n    private IEnumerable<ImportIdentity> ParsedIdentities => GetScope<GetParsedAocSteps>(Identity).Values.Select(aocStep => new ImportIdentity {AocType = aocStep.AocType, Novelty = aocStep.Novelty, DataNode = Identity});",
+                "\n    private IEnumerable<string> rawVariableNovelties => GetStorage().GetRawVariables(Identity).Select(rv => rv.Novelty).Distinct().Concat(Novelties.C.RepeatOnce());",
+                "\n    private IEnumerable<AocStep> calculatedAocSteps => GetStorage().AocConfigurationByAocStep.Values.Where(x => (x.DataType.Contains(DataType.Calculated) || x.DataType.Contains(DataType.CalculatedTelescopic) ) &&",
+                "\n        (!isReinsurance ? !new []{AocTypes.CRU, AocTypes.RCU}.Contains(x.AocType) : true) && rawVariableNovelties.Contains(x.Novelty) ).Select(x => new AocStep(x.AocType, x.Novelty));",
+                "\n    private IEnumerable<ImportIdentity> SpecialIdentities => calculatedAocSteps.Select(x => new ImportIdentity {AocType = x.AocType, Novelty = x.Novelty, DataNode = Identity })",
+                "\n        .Concat(GetStorage().GetAllAocSteps(InputSource.Opening).Select(aocStep => new ImportIdentity {AocType = aocStep.AocType, Novelty = aocStep.Novelty, DataNode = Identity}));//Add BOP,I and IA,I",
```

Comment:
Is this //Add BOP,I and IA,I a TODO?

Reply:
It is an explanation of that line. But it can be understood from the config file. I removed it.